### ISSUE TITLE
Command failed when pulling certificate from device

### ISSUE THREAD

**Issue by dejolt** [facebook/flipper#2871]
## 🐛 Bug Report

  It seems that there's an issues with the certificate pull from the device. 

## To Reproduce

Flipper uses this command to pull the file:
 **idb --log DEBUG file pull --udid (device id) --bundle-id (com.xxxxx.xxxxx) 'Library/Application Support /sonar/app.csr' 'destination folder'.**

If I use the same command in terminal, the command does not work and fails. 

But if use the same command and add a backslash ( **\\** ) at the end of the **Application** like below, the file gets copied to the destination folder. Aka the certificate gets copied and there are no IDB errors.
 **idb --log DEBUG file pull --udid (device id) --bundle-id (com.xxxxx.xxxxx) 'Library/Application\ Support /sonar/app.csr' 'destination folder'.**

Thanks for reviewing it!

## Environment

  Mac OS Big Sur 11.6
  Iphone X iOS 14.8
  Flipper 0.109.0
  React Native 0.65.1

**Comment by mweststrate** [comment]
I suspect your idb / idb_companion is outdated. Which version is reported if you run `idb_companion`? Could you update to the latest version?

cc @passy

**Comment by dejolt** [comment]
> I suspect your idb / idb_companion is outdated. Which version is reported if you run `idb_companion`? Could you update to the latest version?
> 
> cc @passy

Well my idb-companion/idb version is 1.1.5. I even tried with source builds of idb and still the same error.
The interesting part is that, if I modify the Flipper source and build it like I mentioned above by adding **/app.csr**, I got no problems. But if I download a built version of Flipper I get a certificate error. I'm assuming is something on my machine that is wrong, since nobody else complained about it, but can't figure it out what's wrong.

Thanks.

**Comment by mweststrate** [comment]
Something did recently change here, so it could have been introduced by D30541778 / https://github.com/facebook/flipper/commit/89d7e17c260220e4ac96e03583388af8d652e6b6

**Comment by dejolt** [comment]
> Something did recently change here, so it could have been introduced by D30541778 / [89d7e17](https://github.com/facebook/flipper/commit/89d7e17c260220e4ac96e03583388af8d652e6b6)

I think you're right, because it's exactly what's happening. The fix you mentioned, specifies that the destination file name is omitted (app.csr) and it's exactly what I was adding to fix the problem.

**Comment by mweststrate** [comment]
@passy would you mind taking a closer look at this in time?

**Comment by passy** [comment]
Thanks for the detailed notes! That seems like a pretty clear cut bug. I'll take a look next week.

**Comment by mweststrate** [comment]
@passy seems there is some more background linked here: D27590885.

**Comment by dejolt** [comment]
> Hey @passy, any update on this? I can confirm what @dejolt said, the command does indeed work if you append /app.csr to the end of the destination argument.

Yeah so all you got to do is append **/app.csr** to the end of the command in the file **iOSContainerUtility.tsx** on line **203** (just like I pointed out in the image in the bug report). You can find the file by downloading the source code of Flipper 0.113.0, unzip, then the path of the file is desktop/app/src/server/devices/iOS/iOSContainerUtility.tsx. And then you build from source. Hope it helps!

**Comment by passy** [comment]
Apologies for the delays on this. It's been a few busy weeks. Sadly this week doesn't look much better but if nobody else has a chance to look into this, I should be able to pick this up early next week.

**Comment by tomasmozeris** [comment]
> Indeed reverting to version `0.104.0` fixes this issue for us.

Thanks, had same issue, none of newer versions worked on real iOS device, but downgrading to `0.104.0` works

**Comment by elliottkember** [comment]
I also have this error. Downgrading Flipper doesn't fix it (the change is old), and I am unable to build from source on an M1 mac so I can't change iOSContainerUtility.tsx. I also can't determine my IDB version, but I had to install it under Rosetta homebrew (https://github.com/facebook/idb/issues/649)

**Comment by aigoncharov** [comment]
Fixed in https://github.com/facebook/flipper/commit/9975aa9319e69317bec986f290572ca4e1ca2657
Should be included into the next release

**Comment by hlspablo** [comment]
guys, this is related to idb python client, use pip3.9 install fb-idb, and remove any previous installed version that you have.
